### ISSUE TITLE
InfoGAN model: TypeError: unsupported operand type(s) for *: 'range' and 'int'

### ISSUE THREAD

**Issue by serfaniane** [tensorflow/models#3413]
When using `models/research/gan/tutorial.ipynb` in the InfoGAN Evaluation part I get the following error:
`TypeError: unsupported operand type(s) for *: 'range' and 'int'`

The line that throws the error is:
`cat_noise = tf.constant(range(cat_dim) * (images_to_eval // cat_dim))`

There are many many more errors on this tutorial as well.

**Comment by serfaniane** [comment]
A stupid way to fix it would be:

```output = np.zeros(images_to_eval, dtype=int)
for l in range(images_to_eval):
    output[l] = cat_dim * (l // cat_dim)
    for k in range(cat_dim-1):
        output[l] *= k
cat_noise = tf.constant(output)

**Comment by bignamehyp** [comment]
@joel-shor can you please take a look?

**Comment by aqsasamreen** [comment]
i was also facing this problem while running create_dataset.py  for drawing classifier and i solve it as
reading_order = list(range(len(file_handles))) * int(observations_per_class)

**Comment by joel-shor** [comment]
The tutorial isn't yet python3 compatible. I'll make changes in the next few days go make it so. In the meantime, can you try running the tutorial with python2 and see if it works?

**Comment by joel-shor** [comment]
Should be fixed by @#3630.
